controllers/authpolicy_controller.go

```diff
@@ -167,6 +167,44 @@ func (r *AuthPolicyReconciler) reconcileResources(ctx context.Context, ap *api.A
 		return err
 	}
 
+	// if the AuthPolicy(ap) targets a Gateway then all policies attached to that Gateway need to be checked.
+	// this is due to not knowing if the Gateway AuthPolicy was updated to include or remove the overrides section.
+	switch obj := targetNetworkObject.(type) {
+	case *gatewayapiv1.Gateway:
+		gw := kuadrant.GatewayWrapper{Gateway: obj}
+		annotation, ok := gw.GetAnnotations()[ap.BackReferenceAnnotationName()]
+		if !ok {
+			break
+		}
+		policyKeys := &[]client.ObjectKey{}
+		apKey := client.ObjectKeyFromObject(ap)
+		err = json.Unmarshal([]byte(annotation), policyKeys)
+		if err != nil {
+			return err
+		}
+		for _, policyKey := range *policyKeys {
+			if policyKey == apKey {
+				continue
+			}
+
+			ref := &api.AuthPolicy{}
+			err = r.Client().Get(ctx, policyKey, ref)
+			if err != nil {
+				return err
+			}
+
+			refNetworkObject := &gatewayapiv1.HTTPRoute{}
+			err = r.Client().Get(ctx, ref.TargetKey(), refNetworkObject)
```

Comment:
Maybe add a comment here that this only works because of single policy of a kind allowed per target object. Since the gateway policy was skipped in line 186, all the other refs are of policies targeting routes. In the future, that may no longer be the case.

Reply:
I see your point, but I don't think a comment is worth it. I would assume there are other spots in the code base that assume only one type of resources is attached to the gateway. When the day comes when we support it, I would expect that would be a large change to the whole code base and not just here.